src/valkeymodule.h

```diff
@@ -794,6 +794,16 @@ typedef void (*ValkeyModuleInfoFunc)(ValkeyModuleInfoCtx *ctx, int for_crash_rep
 typedef void (*ValkeyModuleDefragFunc)(ValkeyModuleDefragCtx *ctx);
 typedef void (*ValkeyModuleUserChangedFunc)(uint64_t client_id, void *privdata);
 
+/* Type definitions for implementing scripting engines modules. */
+typedef struct ValkeyModuleScriptingEngineFunctionLibrary ValkeyModuleScriptingEngineFunctionLibrary;
+typedef struct ValkeyModuleScriptingEngineFunctionCallCtx ValkeyModuleScriptingEngineFunctionCallCtx;
+typedef int (*ValkeyModuleCreateScriptingEngineFunc)(void *engine_ctx, ValkeyModuleScriptingEngineFunctionLibrary *li, const char *code, size_t timeout, char **err);
+typedef void (*ValkeyModuleCallScriptingEngineFunctionFunc)(ValkeyModuleScriptingEngineFunctionCallCtx *func_ctx, void *engine_ctx, void *compiled_function, ValkeyModuleString **keys, size_t nkeys, ValkeyModuleString **args, size_t nargs);
```

Comment:
Can we make all of these follow the same pattern with a ValkeyModuleScriptingEngine prefix? Not sure the suggestion is good, but you get the idea:

```suggestion
typedef int (*ValkeyModuleScriptingEngineCreateFunc)(void *engine_ctx, ValkeyModuleScriptingEngineFunctionLibrary *li, const char *code, size_t timeout, char **err);
typedef void (*ValkeyModuleScriptingEngineFunctionCallFunc)(ValkeyModuleScriptingEngineFunctionCallCtx *func_ctx, void *engine_ctx, void *compiled_function, ValkeyModuleString **keys, size_t nkeys, ValkeyModuleString **args, size_t nargs);
```

Reply:
Yes, that was my intention, but for some reason I typed the names in this weird form.